config/docker/fragment/kernelci.jinja2

```diff
@@ -8,12 +8,13 @@ RUN pip3 install --upgrade pip
 ARG core_url=https://github.com/kernelci/kernelci-core.git
 ARG core_rev=main
 RUN apt-get update && apt-get install --no-install-recommends -y git
+RUN pip3 install git+$core_url@$core_rev#egg=kernelci[dev]
```

Comment:
What's the reason for doing this?  One issue here is that the kernelci-core git repository gets cloned twice which may introduce some discrepancy one day between the package installed and the config files.  Please explain in the commit message what difference it makes to use pip3 rather than setup.py.

Reply:
I added kernelci[dev] package definition in setup.py to reflect what is defined in the `requirements.txt` and `requirements-dev.txt`. Using `pip` allows us also to specify local, remote, repo URLs or package index to install the package as well as specify what `extras_require` sections will be installed. 

While I agree that cloning the repository may lead to some discrepancy, I think that the problem is a bit more broad. In general lack of proper versioning leads to a situation in which we cannot be sure if the version of the kernelci package installed by the kernelci fragment is consistent between 2 docker images build one after another. This actually applies to any situation where kernelci-core repository is cloned more than once and the revision is specified as `main`.

At the moment I changed the kernelci fragment definition to clone the repository just once, but I'd treat it as a temporary solution before the docker images versioning is introduced and configuration and application code are kept in separate repositories.